### PR TITLE
Fix expansion for `ValueSet` with no concepts based on CodeSystem `urn:ietf:bcp:13`

### PR DESCRIPTION
**Pull request by codeforgreen**
#5634

Summary of changes:
- updated `CommonCodeSystemsTerminologyService` to return a `CodeSystem` with no concepts and content set to not-present for `urn:ietf:bcp:13`. Previously it was returning null. See: http://hl7.org/fhir/valueset-codesystem-content-mode.html#expansion
- Updated `InMemoryTerminologyServerValidationSupport` to avoid expansion failure for a `ValueSet` based on the above `CodeSystem`, for the case where the expansion is a full expansion.
- Added some test cases. Other common `CodeSystem` types were tested and some additional issues were found. I've captured them in #5643

### CODE REVIEW

File: hapi-fhir-validation/src/main/java/org/hl7/fhir/common/hapi/validation/support/InMemoryTerminologyServerValidationSupport.java [hunk 1]
```diff
@@ -998,15 +996,22 @@ private boolean expandValueSetR5IncludeOrExclude(
...
-					if (isNotBlank(theInclude.getSystem())
-							&& !theInclude.getConcept().isEmpty()
-							&& theInclude.getFilter().isEmpty()
-							&& theInclude.getValueSet().isEmpty()) {
-						theInclude.getConcept().stream()
-								.map(t -> new FhirVersionIndependentConcept(
-										theInclude.getSystem(), t.getCode(), t.getDisplay(), theInclude.getVersion()))
-								.forEach(t -> nextCodeList.add(t));
-						ableToHandleCode = true;
+					boolean isIncludeFromSystem = isNotBlank(theInclude.getSystem())
```
Comment (jmarchionatto) [inline hapi-fhir-validation/src/main/java/org/hl7/fhir/common/hapi/validation/support/InMemoryTerminologyServerValidationSupport.java:999]: What if both valueSet and concept are empty? Maybe the check for ampty concept should stay?

File: hapi-fhir-validation/src/test/java/org/hl7/fhir/common/hapi/validation/support/InMemoryTerminologyServerValidationSupportTest.java [hunk 2]
```diff
@@ -54,8 +50,159 @@ public void before() {
 		myDefaultSupport.fetchCodeSystem("http://foo");
 	}
 
+	@ParameterizedTest
+	@ValueSource(strings = {
+			CommonCodeSystemsTerminologyService.MIMETYPES_VALUESET_URL,
+			CommonCodeSystemsTerminologyService.CURRENCIES_VALUESET_URL,
+			CommonCodeSystemsTerminologyService.LANGUAGES_VALUESET_URL
+			// TODO:  other types cannot be expanded just yet, check if they need to be implemented
```
Comment (jmarchionatto) [inline hapi-fhir-validation/src/test/java/org/hl7/fhir/common/hapi/validation/support/InMemoryTerminologyServerValidationSupportTest.java:58]: I think this TODO needs to be answered before approval.
Please look for an answer and add it as a comment

File: hapi-fhir-validation/src/test/java/org/hl7/fhir/common/hapi/validation/support/InMemoryTerminologyServerValidationSupportTest.java [hunk 3]
```diff
@@ -54,8 +50,159 @@ public void before() {
...
+		assertNotNull(valueSet.getExpansion());
+	}
+
+
+	@ParameterizedTest
+	@ValueSource(strings = {
+			CommonCodeSystemsTerminologyService.MIMETYPES_CODESYSTEM_URL,
+			CommonCodeSystemsTerminologyService.COUNTRIES_CODESYSTEM_URL,
+			CommonCodeSystemsTerminologyService.CURRENCIES_CODESYSTEM_URL
+			// TODO: other types cannot be expanded just yet, check if they need to be implemented
```
Comment (jmarchionatto) [inline hapi-fhir-validation/src/test/java/org/hl7/fhir/common/hapi/validation/support/InMemoryTerminologyServerValidationSupportTest.java:80]: I think this TODO needs to be answered before approval.
Please look for an answer and add it as a comment

**Review by jmarchionatto — CHANGES_REQUESTED** [review]
Added a few comments. Also, I would like to have the pointed TODO questions replied and responses added as comments, before approving

File: hapi-fhir-validation/src/main/java/org/hl7/fhir/common/hapi/validation/support/InMemoryTerminologyServerValidationSupport.java (hunk 1 above)
Comment (codeforgreen) [reply hapi-fhir-validation/src/main/java/org/hl7/fhir/common/hapi/validation/support/InMemoryTerminologyServerValidationSupport.java:999]: I did leave it please see the if within this if which uses isIncludeWithDeclaredConcepts. All 4 parts of the conditin are kept. Basically clubbed together some parts of the condition to reuse and make the code easier to read. So my intention here is only to add an else and do some naming to the existing condition, but no change of behaviour to existing conditions.
Let me know if this makes sense.

File: hapi-fhir-validation/src/test/java/org/hl7/fhir/common/hapi/validation/support/InMemoryTerminologyServerValidationSupportTest.java (hunk 2 above)
Comment (codeforgreen) [reply hapi-fhir-validation/src/test/java/org/hl7/fhir/common/hapi/validation/support/InMemoryTerminologyServerValidationSupportTest.java]: I've removed the TODOs and captured them here: https://github.com/hapifhir/hapi-fhir/issues/5643. Let me know if anything else or more detail is needed.

File: hapi-fhir-validation/src/test/java/org/hl7/fhir/common/hapi/validation/support/InMemoryTerminologyServerValidationSupportTest.java (hunk 3 above)
Comment (codeforgreen) [reply hapi-fhir-validation/src/test/java/org/hl7/fhir/common/hapi/validation/support/InMemoryTerminologyServerValidationSupportTest.java]: I've removed the TODOs and captured them here: https://github.com/hapifhir/hapi-fhir/issues/5643. Let me know if anything else or more detail is needed.

File: hapi-fhir-validation/src/main/java/org/hl7/fhir/common/hapi/validation/support/InMemoryTerminologyServerValidationSupport.java (hunk 1 above)
Comment (jmarchionatto) [reply hapi-fhir-validation/src/main/java/org/hl7/fhir/common/hapi/validation/support/InMemoryTerminologyServerValidationSupport.java:999]: It does. Than you

**Review by jmarchionatto — APPROVED** [review]
Looks good. Approved.

**Pull request merged**